sanic/response.py

```diff
@@ -112,8 +117,17 @@ async def send(
         """
         if data is None and end_stream is None:
             end_stream = True
-        if end_stream and not data and self.stream.send is None:
-            return
+        if self.stream is None:
+            raise SanicException(
+                "Stream was disconnected with this response instance."
+            )
```

Comment:
Let's make this also `ServerError`. Also, I am not sure that the message is helpful. How is the user supposed to know which response is "this"? Can we just say "Stream is disconnected"?

Reply:
How about "No stream is connected to the response object"? This error message is raised in `Response` class, so it is related to the response, right?

Reply:
That works.

Reply:
Thanks, fixed.